data/modules/CargoRun/CargoRun.lua

```diff
@@ -0,0 +1,1148 @@
+table.insert(consumer_goods, spaceship_parts)
+
+local expensive = {} -- price >= 175
+table.insert(expensive, art_objects)
+table.insert(expensive, diamonds)
+table.insert(expensive, neptunium)
+table.insert(expensive, plutonium)
+table.insert(expensive, spaceship_parts)
+table.insert(expensive, uranium)
+
+local crazy = {}
+table.insert(crazy, quibbles)			-- 1
+table.insert(crazy, wedding_dresses)		-- 2
+
+local custom_cargo = {}
+table.insert(custom_cargo, chemical)		-- 1
+table.insert(custom_cargo, mining)		-- 2
+table.insert(custom_cargo, hardware)		-- 3
+table.insert(custom_cargo, infrastructure)	-- 4
+table.insert(custom_cargo, consumer_goods)	-- 5
+table.insert(custom_cargo, expensive)		-- 6
+table.insert(custom_cargo, crazy)		-- 7
+
+local HABER = 2 -- INTROTEXT_WHOLESALER_2
+
+local ads = {}
+local missions = {}
+
+local isQualifiedFor = function(reputation, ad)
+	return
+		reputation >= 12 or
+		ad.localdelivery or
+		(ad.risk <  0.1 and ad.urgency <= 0.1) or
+		(ad.risk <  0.5 and ad.urgency <= 0.5 and reputation >= 4) or
+		false
+end
+
+local getNumberOf = function (str)
+	local num = 0
+
+	while l[str .. "_" .. num + 1] do
+		num = num + 1
+	end
+	return num
+end
+
+local onChat = function (form, ref, option)
+	local ad = ads[ref]
+	local num, introtext, cargo_picked_up
+
+	form:Clear()
+
+	if option == -1 then
+		form:Close()
+		return
+	end
+
+	local qualified = isQualifiedFor(Character.persistent.player.reputation, ad)
+
+	form:SetFace(ad.client)
+
+	if not qualified then
+		form:SetMessage(l["DENY_" .. Engine.rand:Integer(1, getNumberOf("DENY"))])
+		return
+	end
+
+	if option == 0 then
+		if ad.localdelivery then
+			if ad.pickup then
+				introtext = l["INTROTEXT_PICKUP_LOCAL_" .. ad.introtext]
+			else
+				introtext = l["INTROTEXT_LOCAL_" .. ad.introtext]
+			end
+		elseif ad.wholesaler then
+			introtext = l["INTROTEXT_WHOLESALER_" .. ad.introtext]
+		elseif ad.pickup then
+			introtext = l["INTROTEXT_PICKUP_" .. ad.introtext]
+		else
+			introtext = l["INTROTEXT__" .. ad.branch .. "__" .. ad.cargotype] or l["INTROTEXT__" .. ad.branch] or l["INTROTEXT_" .. ad.introtext]
+		end
+		local introtext = string.interp(introtext, {
+			name		= ad.client.name,
+			cash		= Format.Money(ad.reward),
+			cargoname	= l[custom_cargo[ad.branch][ad.cargotype].l10n_key],
+			starport	= ad.location:GetSystemBody().name,
+			system		= ad.location:GetStarSystem().name,
+			sectorx		= ad.location.sectorX,
+			sectory		= ad.location.sectorY,
+			sectorz		= ad.location.sectorZ,
+			dom_starport	= ad.domicile:GetSystemBody().name,
+			dom_system	= ad.domicile:GetStarSystem().name,
+			dom_sectorx	= ad.domicile.sectorX,
+			dom_sectory	= ad.domicile.sectorY,
+			dom_sectorz	= ad.domicile.sectorZ,
+			dist	  = string.format("%.2f", ad.dist),
+		})
+		form:SetMessage(introtext)
+
+	elseif option == 1 then
+		local whysomuchtext = l["WHYSOMUCHTEXT__" .. ad.branch .. "__" .. ad.cargotype] or l["WHYSOMUCHTEXT__" .. ad.branch]
```

Comment:
If I'm to nitpick the slightly ambiguous `whysomuchtext / WHYSOMUCHTEXT_` variable and string names are not consistent with other string variable names like `howmuch`, I'd opt for just `whysomuch`.